src/OrleansRuntime/ReminderService/LocalReminderService.cs

```diff
@@ -201,17 +203,28 @@ public Task Stop()
         private async Task ReadAndUpdateReminders()
         {
             if (stoppedCancellationTokenSource.IsCancellationRequested) return;
+            localRangeSequence++;
+            var cachedLocalRangeSequence = localRangeSequence;
 
             // try to retrieve reminder from all my subranges
-            myRange = ring.GetMyRange();
-            if (logger.IsVerbose2) logger.Verbose2("My range= {0}", myRange);
+            if (logger.IsVerbose2) logger.Verbose2($"My range= {myRange}, LocalRangeSequence{localRangeSequence}");
+
+            foreach (var reminder in localReminders.Where(r => !myRange.InRange(r.Key.GrainRef)).Select(r => r.Value))
+            {
+                if (logger.IsVerbose2)
+                    logger.Verbose2("Not in my range anymore, so removing. {0}", reminder);
+                // remove locally
+                reminder.StopReminder(logger);
+                localReminders.Remove(reminder.Identity);
```

Comment:
You sure its OK to remove from localReminders while iterating over it? Usualy not, here due to LINQ maybe? Not sure.